### PR TITLE
Added unit tests for RecordUtil

### PR DESCRIPTION
**Pull request by BashCache**
[OpenForce 2022] Add unit test for RecordUtil #15699

Changes proposed in this pull request:
- I have added unit test for the 3 methods in RecordUtil class.
- Built the project and unit tests passed with 100% code coverage.

### CODE REVIEW

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import java.util.Collections;
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+
+public final class RecordUtilTest {
+    @Test
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:16]: Please add blank line.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    public void assertExtractPrimaryColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        List<Column> actual = RecordUtil.extractPrimaryColumns(dataRecord);
+        assertThat(actual.size(), is(2));
+    }
+
+    @Test
+    public void assertExtractUpdatedColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        List<Column> actual = RecordUtil.extractUpdatedColumns(dataRecord);
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:26]: `Collection` may be better.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+
+public final class RecordUtilTest {
+    @Test
+    public void assertExtractPrimaryColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        List<Column> actual = RecordUtil.extractPrimaryColumns(dataRecord);
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:19]: `Collection` may be better.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    public void assertExtractUpdatedColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        List<Column> actual = RecordUtil.extractUpdatedColumns(dataRecord);
+        assertThat(actual.size(), is(3));
+    }
+
+    @Test
+    public void assertExtractConditionalColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        List<Column> actual = RecordUtil.extractConditionColumns(dataRecord, Collections.singleton("c1"));
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:33]: `Collection` may be better.

**Review by totalo — CHANGES_REQUESTED** [review]

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.junit.Assert.assertThat;
+
+public final class RecordUtilTest {
+
+    @Test
+    public void assertExtractPrimaryColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        Collection<Column> actual = RecordUtil.extractPrimaryColumns(dataRecord);
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:36]: There is one more problem, in addition to the number of elements, we'd better check the specific content.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        assertThat(actual.size(), is(2));
+    }
+
+    @Test
+    public void assertExtractConditionalColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        Collection<Column> actual = RecordUtil.extractConditionColumns(dataRecord, Collections.singleton("c1"));
+        assertThat(actual.size(), is(3));
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:44]: There is one more problem, in addition to the number of elements, we'd better check the specific content.

File: shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+    }
+
+    @Test
+    public void assertExtractUpdatedColumns() {
+        DataRecord dataRecord = mockDataRecord("t2");
+        Collection<Column> actual = RecordUtil.extractUpdatedColumns(dataRecord);
+        assertThat(actual.size(), is(3));
```
Comment (totalo) [inline shardingsphere-test/shardingsphere-pipeline-test/src/test/java/org/apache/shardingsphere/data/pipeline/core/record/RecordUtilTest.java:51]: The same as above.

**Review by totalo — CHANGES_REQUESTED** [review]

**Review by totalo — CHANGES_REQUESTED** [review]

**Comment by totalo** [comment]
Thanks for the contribution, I'll merge it when the ci is passed. Looking forward to your more contributions in this community.

**Comment by BashCache** [comment]
Thank you @totalo for helping me out. Made my first contribution for apache sharding-sphere. Looking forward to cotribute more :)

**Comment by totalo** [comment]
> Thank you @totalo for helping me out. Made my first contribution for apache sharding-sphere. Looking forward to cotribute more :)

Goob job!Merged.

**Review by totalo — APPROVED** [review]

**Pull request merged**